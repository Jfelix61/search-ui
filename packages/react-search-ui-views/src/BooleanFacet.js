import PropTypes from "prop-types";
import React from "react";

import { FacetValue, FilterValue } from "./types";
import { appendClassName, getFilterValueDisplay } from "./view-helpers";

function BooleanFacet({
  className,
  label,
  options,
  onChange,
  onRemove,
  onSelect,
  values
}) {
  // const trueOptions = options.map(option => option.value === "true")
  const trueCount = options.find(option => option.value === "true").count;
  const isSelected = values.includes("true");

  const isSelectedClass = isSelected ? "selected" : "";

  const apply = () => onChange("true");

  const remove = () => onRemove("true");

  return (
    <fieldset className={appendClassName("sui-facet", className)}>
      <legend className="sui-facet__title">{label}</legend>

      <div className="sui-boolean-facet">
        {options.map(option => {
          // console.log(option);
          const checked = values.includes("true");
          // const checked = !!values.find(value => {
          //   if (value && value.name && option.value.name === value.name)
          //     return true;
          //   if (deepEqual(option.value, value)) return true;
          //   return false;
          // });

          return (
            <div
              key={`${getFilterValueDisplay(option.value)}`}
              className={"sui-boolean-facet__option-input-wrapper"}
              onClick={apply}
            >
              <label className="sui-boolean-facet__option-label">
                <div className="sui-boolean-facet__option-input-wrapper">
                  <input
                    id={`example_facet_${label}${getFilterValueDisplay(
                      option.value
                    )}`}
                    className="sui-boolean-facet__checkbox"
                    type="checkbox"
                    name="checkbox"
                    value="value"
                    checked={checked}
                    onChange={() =>
                      checked ? onRemove("true") : onSelect("true")
                    }
                  />
                  <span className="sui-boolean-facet__input-text">
                    {`${getFilterValueDisplay(option.value)}`}
                  </span>
                </div>
                <span className="sui-boolean-facet__option-count">
                  {option.count.toLocaleString("en")}
                </span>
              </label>
            </div>
          );
        })}
        <div
          className={appendClassName(isSelectedClass, className)}
          onClick={apply}
        >
          {label}{" "}
          {!isSelectedClass ? (
            `(${trueCount})`
          ) : (
            <span onClick={remove}>X</span>
          )}
        </div>
      </div>
    </fieldset>
  );
}

BooleanFacet.propTypes = {
  className: PropTypes.string,
  label: PropTypes.string.isRequired,
  onRemove: PropTypes.func.isRequired,
  onSelect: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(FacetValue).isRequired,
  values: PropTypes.arrayOf(FilterValue).isRequired,
  onChange: PropTypes.func.isRequired
};

export default BooleanFacet;
