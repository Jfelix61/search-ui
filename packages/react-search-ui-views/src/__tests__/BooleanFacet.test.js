import React from "react";
import BooleanFacet from "../BooleanFacet";
import { shallow } from "enzyme";

const params = {
  label: "A Facet",
  onRemove: jest.fn(),
  onChange: jest.fn(),
  options: [
    {
      value: "true",
      count: 10
    }
  ],
  values: []
};

it("renders", () => {
  const wrapper = shallow(<BooleanFacet {...params} />);
  expect(wrapper).toMatchSnapshot();
});

describe("determine selection", () => {
  it("will show count when Facet is not selected", () => {
    const wrapper = shallow(<BooleanFacet {...params} />);

    expect(wrapper.exists("span")).toBe(false);
  });

  it("will show X when Facet is selected", () => {
    const wrapper = shallow(<BooleanFacet {...params} />);
    wrapper.setProps({ values: ["true"] });

    expect(wrapper.exists("span")).toBe(true);
  });

  it("will show count once Facet is clicked twice", () => {
    const wrapper = shallow(<BooleanFacet {...params} />);

    //First click
    wrapper.setProps({ values: ["true"] });

    expect(wrapper.exists("span")).toBe(true);

    //Second click
    wrapper.setProps({ values: [] });
    expect(wrapper.exists("span")).toBe(false);
  });

  it("shows class 'selected' when Facet is selected", () => {
    const wrapper = shallow(<BooleanFacet {...params} />);

    wrapper.setProps({ values: ["true"] });

    expect(wrapper.find("div").props().className).toEqual("selected");
  });
});

it("renders with className prop applied", () => {
  const customClassName = "test-class";
  const wrapper = shallow(
    <BooleanFacet className={customClassName} {...params} />
  );
  const { className } = wrapper.props();
  expect(className).toEqual("test-class");
});
