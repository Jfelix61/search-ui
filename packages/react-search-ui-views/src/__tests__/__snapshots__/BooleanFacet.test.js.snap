// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className=""
  onClick={[Function]}
>
  A Facet
   
  (10)
</div>
`;
